src/mcts/utils.h

```diff
@@ -207,7 +249,7 @@ class MctsResult {
   void add(Action a, float visits) {
     if (mctsPolicy.size() <= (size_t)a) {
       if (mctsPolicy.capacity() <= (size_t)a) {
-        mctsPolicy.reserve(mctsPolicy.size() * 2);
+        mctsPolicy.reserve(mctsPolicy.size() * 2); // bad: this assumes that a is not too big
```

Comment:
`.reserve()` only has performance implications, it just allocates array space in advance.

If you're actually concerned about a being "not too big" you should add an assert.

Exponential allocation ("if I exceed the available array space in a `std::vector`, I'm going to move it to another memory location with twice the space") is a pretty common strategy.

Reply:
Thanks. And urgh, I thought I had deleted this comment. I initially thought it was doing something incorrect (as you can see), but later understood it was not (yet the comment stayed in).